lua/ts_context_commentstring/utils.lua

```diff
@@ -156,24 +125,22 @@ function M.get_node_at_cursor_start_of_line(only_languages, location)
   }
 
   -- Get the language tree with nodes inside the given range
-  local root = get_parser()
+  local root = vim.treesitter.get_parser()
   local language_tree = language_for_range(root, range, only_languages)
 
   -- Get the sub-tree of the language tree that contains the given range.
   -- If there are multiple trees in the buffer for the same injected language,
   -- then we need to make sure that we are operating on the correct tree.
-  local tree = vim.tbl_filter(function(tree)
-    return tree_contains(tree, range)
-  end, language_tree:trees())[1]
+  local ltree = vim.tbl_filter(function(tree)
+    return tree:contains(range)
+  end, language_tree:children())[1]
```

Comment:
This breaks if there are no relevant children in the tree. It looks like previously, the language tree itself would also be included in `:trees()`, but `:children()` only returns the actual children. So, if we have a `tsx` tree with no children, the children and `ltree` would be empty, and this function would return early.

To reproduce, have a `.tsx` file with the following contents and try to comment the `<h1>` line:

```tsx
(
  <div>
    <h1>Hello</h1>
  </div>
)
```

It will comment with `// %s`, but it should be `{/* %s */}`.

One way to fix it would be to default this to `language_tree`:

```suggestion
  end, language_tree:children())[1] or language_tree
```

Reply:
Oh, this is really a great example. There's a bug on `main` when you try to comment and uncomment `<div>`. Similarly, the same bug appears if you try to comment and uncomment all three lines containing `<div>`.

This boils down to `{/* */}` being not a comment, but an object wrapping a comment.

I believe I came up with a working, if hacky, solution by adding `object = '{/* %s */}',` config to TSX defaults.

Reply:
I think that you mean this issue: https://github.com/JoosepAlviste/nvim-ts-context-commentstring/issues/29 which is a bit more difficult to solve. With the current fix, this breaks, for example:

```ts
const a = {
    |hello: 'world',
    ^ cursor, commenting
}
```

now results in

```ts
const a = {
    {/* hello: 'world', */}
}
```

I don't think we should tackle that issue in this PR.

I just meant this case:

```tsx
(
  <div>
    |<h1>Hello</h1>
    ^ cursor, commenting
  </div>
)
```

results in

```tsx
(
  <div>
    // <h1>Hello</h1>
  </div>
)
```

which should be solved by the original suggestion I made (to add `or language_tree`)

Reply:
Oof, you're right, best not to tackle this here. Especially with me not working with TSX files currently.

I removed the "fix".